src/Models/ApiKey.php

```diff
@@ -91,4 +84,15 @@ public function markAsUsed()
             'last_used_at' => $this->freshTimestamp()
         ])->save();
     }
+
+    public function scopeOfKey(Builder $query, string $key): Builder
+    {
+        if (strpos($key, '|') === false) {
+            return $query->where('key', hash('sha256', $key));
+        }
+
+        [$id, $key] = explode('|', $key, 2);
+
+        return $query->where('id', $id)->where('key', hash('sha256', $key));
+    }
```

Comment:
Isn't this is a breaking change now that we're only querying for hashed keys? All existing keys in someone's database would be no longer be good without migrating them to a hashed key first.

Reply:
Yeah, most of these changes here are breakign changes, that's why I suggested we should release a new major version.

So you guys wanna introduce these changes without releasing a new major version, 3.0?

Reply:
Totally agree on the upgrade guide and the command to hash all existing key, I will work on those.

Reply:
I don't mind introducing breaking changes, just as long as we have proper documentation and offer a smooth transition for the people that use our package.

For example, we obviously want to keep this query performant, but maybe we could add a config flag or static method that could be set/called to allow backwards compatibility with non-hashed keys. For example, something like `Keyable::allowNonHashedDBKeys()` could be called from AppServiceProvider while they work on hashing all of their existing keys. Then if that is called, we can conditionally have a fallback condition in this scope.